vulture_os/toolkit/api_parser/perception_point_x_ray/perception_point_x_ray.py

```diff
@@ -0,0 +1,226 @@
+#!/home/vlt-os/env/bin/python
+"""This file is part of Vulture OS.
+
+Vulture OS is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+__author__ = "nlancon"
+__credits__ = []
+__license__ = "GPLv3"
+__version__ = "4.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'PERCEPTION POINT X RAY API Parser'
+__parser__ = 'PERCEPTION_POINT_X_RAY'
+
+import logging
+from django.conf import settings
+from django.utils import timezone
+
+from toolkit.api_parser.api_parser import ApiParser
+
+from requests import Session, exceptions
+from json import dumps, decoder
+
+from datetime import datetime, timedelta
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+
+class PerceptionPointXRayAPIError(Exception):
+    pass
+
+
+class PerceptionPointXRayParser(ApiParser):
+
+    HEADERS = {
+        "Accept": "application/json"
+    }
+
+    LOG_TYPES = ["SUS", "MAL", "SPM"]  # Malicious, spam and suspicious
+
+    def __init__(self, data):
+        super().__init__(data)
+
+        self.perception_point_x_ray_host = "https://" + data["perception_point_x_ray_host"].split("://")[-1].rstrip()
+        self.token = data['perception_point_x_ray_token']
+
+        self.session = None
+
+    def _execute_query(self, url, query=None, timeout=10):
+
+        retry = 1
+        while retry <= 3:  # 3 retries max
+            if self.session is None:
+                self._connect()
+
+            try:
+                response = self.session.get(
+                    url,
+                    params=query,
+                    timeout=timeout,
+                    proxies=self.proxies,
+                    verify=self.api_parser_custom_certificate or self.api_parser_verify_ssl
+                )
+            except exceptions.Timeout:
```

Comment:
why did you remove the logger?
I just advised you change the log level

Reply:
check line 77

Reply:
I was talking about the log to print the query URL, the parameters and the number of retries

Reply:
Voilà c'est remis